core/state/statedb.go

```diff
@@ -211,7 +212,12 @@ func (self *StateDB) UpdateStateObject(stateObject *StateObject) {
 		self.db.Put(stateObject.CodeHash(), stateObject.code)
 	}
 	addr := stateObject.Address()
-	self.trie.Update(addr[:], stateObject.RlpEncode())
+	data, err := rlp.EncodeToBytes(stateObject)
+	if err != nil {
+		glog.Errorf("can't encode object at %x: %v", addr[:], err)
+		return
```

Comment:
Wondering whether we should `Fatalf` here instead? I know that this code is equivalent to te old behavior, however I'm not sure we should allow progressing and inserting `nil` into the trie if we have an issue with encoding.